subprojects/core-api/src/main/java/org/gradle/api/artifacts/repositories/UrlArtifactRepository.java

```diff
@@ -27,24 +35,76 @@ public interface UrlArtifactRepository {
 
     /**
      * The base URL of this repository.
+     */
+    @ReplacesEagerProperty(replacedAccessors = {
+        @ReplacedAccessor(originalType = URI.class, value = ReplacedAccessor.AccessorType.GETTER, name = "getUrl"),
```

Comment:
It's hard to do since fileResolver is not available in this package. Should we introduce the dependency? 

All our `UrlArtifactRepository` descendants do have an adapter already and any custom descendants should adapt, right? I'd say it's not worth migrating